llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv32/legalize-add.mir

```diff
@@ -181,3 +181,330 @@ body:             |
     PseudoRET implicit $x10, implicit $x11, implicit $x12
 
 ...
+---
+name:  test_nxv1s8
+body:   |
```

Comment:
I think I would prefer a separate test file in test/CodeGen/RISCV/GlobalISel/legalizer/rvv/ that has RUN lines for both rv32 and rv64. That way we don't duplicate identical test content in two different files.

Reply:
I saw you reorganized the testing directory.

> I think I would prefer a separate test file in test/CodeGen/RISCV/GlobalISel/legalizer/rvv/ 

you mean I should create a new folder `rvv/` on top your recent commits?
I don't have a strong opinion regarding where to place vector tests, [AArch64 mixed scalar and vector together](https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/AArch64/GlobalISel/legalize-add.mir) by the way.

Reply:
> I saw you reorganized the testing directory.
> 
> > I think I would prefer a separate test file in test/CodeGen/RISCV/GlobalISel/legalizer/rvv/
> 
> you mean I should create a new folder `rvv/` on top your recent commits? I don't have a strong opinion regarding where to place vector tests, [AArch64 mixed scalar and vector together](https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/AArch64/GlobalISel/legalize-add.mir) by the way.

Yes. We have separate tests for rv32 and rv64 because the scalar arguments and returns values have different types(s32 vs s64) so the tests aren't the same. AArch64 doesn't have this issue. Vector tests for RISC-V could be identical for rv32 and rv64 as long as they don't contain scalar arguments/returns. So I think its better to have a single test file that has 2 RUN lines so that we don't duplicate identical content in 2 different files.

Reply:
So like we'll have `test/CodeGen/RISCV/GlobalISel/legalizer/rvv/add.mir`:
```mir
# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
# RUN: llc -mtriple=riscv32 -mattr=+v -run-pass=legalizer %s -o - | FileCheck %s
# RUN: llc -mtriple=riscv64 -mattr=+v -run-pass=legalizer %s -o - | FileCheck %s
---
name:  test_nxv1s8
body:   |
  bb.0.entry:
    ; CHECK-LABEL: name: test_nxv1s8
    ; CHECK: [[COPY:%[0-9]+]]:_(<vscale x 1 x s8>) = COPY $v8
    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(<vscale x 1 x s8>) = COPY $v9
    ; CHECK-NEXT: [[ADD:%[0-9]+]]:_(<vscale x 1 x s8>) = G_ADD [[COPY]], [[COPY1]]
    ; CHECK-NEXT: PseudoRET implicit [[ADD]](<vscale x 1 x s8>)
    %0:_(<vscale x 1 x s8>) = COPY $v8
    %1:_(<vscale x 1 x s8>) = COPY $v9
    %2:_(<vscale x 1 x s8>) = G_ADD %0, %1
    PseudoRET implicit %2
...
---
name:  test_nxv2s8
body:   |
  bb.0.entry:
    ; CHECK-LABEL: name: test_nxv2s8
    ; CHECK: [[COPY:%[0-9]+]]:_(<vscale x 2 x s8>) = COPY $v8
    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(<vscale x 2 x s8>) = COPY $v9
    ; CHECK-NEXT: [[ADD:%[0-9]+]]:_(<vscale x 2 x s8>) = G_ADD [[COPY]], [[COPY1]]
    ; CHECK-NEXT: PseudoRET implicit [[ADD]](<vscale x 2 x s8>)
    %0:_(<vscale x 2 x s8>) = COPY $v8
    %1:_(<vscale x 2 x s8>) = COPY $v9
    %2:_(<vscale x 2 x s8>) = G_ADD %0, %1
    PseudoRET implicit %2
...
more tests
```
?

Reply:
Yep. We'll probably also need some tests with -mattr=+zve32x instead of -mattr=+v to make sure the nxv*s8/s16/s32 types work when hasVInstructionsI64() is false. But that would have to be a separate file without any nxv*s64. Maybe we only test 1 opcode?